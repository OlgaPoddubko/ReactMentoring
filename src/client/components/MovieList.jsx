// @flow
import * as React from 'react';
import Movie from './Movie';
import { connect } from 'react-redux';

type Props = {
  movies: [],
}

export class MovieList extends React.Component<Props> {
  static defaultProps = {
    movies: [],
  };

  render() {
    const movies = this.props.movies;
    return (
      <div className="items">
        {movies.map(elem =>
          <Movie
            key={elem.id}
            movie={elem}
          />
        )}
        <style jsx>{`
          .items {
            max-width: 1200px;
            margin: 0 auto;
            padding-bottom: 90px;
          }
        `}</style>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  movies: state.gallery.movies,
});

export default connect(mapStateToProps)(MovieList);
