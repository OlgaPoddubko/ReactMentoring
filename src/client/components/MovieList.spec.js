import React from 'react';
import { MovieList } from './MovieList';

describe('render MovieList', () => {
  it('should render MovieList', () => {
    const wrapper = shallow(<MovieList />);
    expect(wrapper).toMatchSnapshot();
  });
});
