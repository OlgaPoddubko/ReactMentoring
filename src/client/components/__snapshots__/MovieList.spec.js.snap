// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MovieList should render MovieList 1`] = `
<div
  className="items"
>
  <style
    jsx={true}
  >
    
          .items {
            max-width: 1200px;
            margin: 0 auto;
            padding-bottom: 90px;
          }
        
  </style>
</div>
`;
